src/qtism/data/storage/xml/XmlResultDocument.php

```diff
@@ -73,4 +98,28 @@ protected function decorateRootElement(DOMElement $rootElement)
         $rootElement->setAttributeNS('http://www.w3.org/2000/xmlns/', 'xmlns:xsi', 'http://www.w3.org/2001/XMLSchema-instance');
         $rootElement->setAttributeNS('http://www.w3.org/2001/XMLSchema-instance', 'xsi:schemaLocation', "http://www.imsglobal.org/xsd/imsqti_${qtiSuffix} ${xsdLocation}");
     }
+
+    protected function inferVersion()
+    {
+        $document = $this->getDomDocument();
+        $root = $document->documentElement;
+        $version = false;
+
+        if (empty($root) === false) {
+            $rootNs = $root->namespaceURI;
+
+            if ($rootNs === 'http://www.imsglobal.org/xsd/imsqti_result_v2p1') {
+                $version = '2.1.0';
+            } elseif ($rootNs === 'http://www.imsglobal.org/xsd/imsqti_result_v2p2') {
+                $version = '2.2.0';
+            }
```

Comment:
Again, IMO doing a return immediately and throwing exception at the most bottom would improve readability

Reply:
Done in next PR... :)